src/concrete/ml/sklearn/base.py

```diff
@@ -1281,13 +1290,23 @@ def __init_subclass__(cls):
                 _TREE_MODELS.add(cls)
                 _ALL_SKLEARN_MODELS.add(cls)
 
-    def __init__(self, n_bits: int):
+    def __init__(self, n_bits: Union[int, Dict[str, int]]):
         """Initialize the TreeBasedEstimatorMixin.
 
         Args:
-            n_bits (int): The number of bits used for quantization.
+            n_bits (int, Dict[str, int]): Number of bits to quantize the model. If an int is passed
+                for n_bits, the value will be used for quantizing inputs and leaves. If a dict is
+                passed, then it should contain "op_inputs" and "op_leaves" as keys with
+                corresponding number of quantization bits so that:
+                    - op_inputs (mandatory): number of bits to quantize the input values
+                    - op_leaves (optional): number of bits to quantize the leaves
+                Default to 6.
         """
-        self.n_bits: int = n_bits
+
+        # Check if 'n_bits' is a valid value
+        _inspect_tree_n_bits(n_bits)
+
+        self.n_bits: Union[int, Dict[str, int]] = n_bits
```

Comment:
why not using `_get_n_bits_dict_trees` directly here (and then remove the use of `_inspect_tree_n_bits(n_bits)` ?

Reply:
Because I got this error: `RuntimeError: Cannot clone object DecisionTreeClassifier(n_bits=3), as the constructor either does not set or modifies parameter n_bits` in test_pipline

Reply:
I assumed it was because of this error, that hadn't put it in the init

Reply:
ah yes I see, pretty inconvenient convention from sklearn : https://stackoverflow.com/questions/24510510/python-scikit-learn-cannot-clone-object-as-the-constructor-does-not-seem-to ..

Reply:
thanks !